std/file.d

```diff
@@ -1731,23 +1752,35 @@ bool attrIsSymlink(uint attributes) @safe pure nothrow @nogc
  * Change directory to $(D pathname).
  * Throws: $(D FileException) on error.
  */
-void chdir(in char[] pathname) @safe
+void chdir(R)(R pathname)
+    if (isInputRange!R && isSomeChar!(ElementEncodingType!R))
 {
+    // Place outside of @trusted block
+    auto pathz = pathname.tempCString!FSChar();
+
     version(Windows)
     {
-        static auto trustedSetCurrentDirectoryW(in char[] path) @trusted
+        static auto trustedSetCurrentDirectoryW(const(FSChar)* pathz) @trusted
         {
-            return SetCurrentDirectoryW(path.tempCStringW());
+            return SetCurrentDirectoryW(pathz);
         }
-        cenforce(trustedSetCurrentDirectoryW(pathname), pathname);
+        static if (isNarrowString!R && is(Unqual!(ElementEncodingType!R) == char))
```

Comment:
Same as https://github.com/D-Programming-Language/phobos/pull/3486 - static if can be hoisted out of version blocks.